core/src/main/java/bisq/core/dao/state/model/DaoState.java

```diff
@@ -145,6 +154,10 @@ private DaoState(int chainHeight,
         this.paramChangeList = paramChangeList;
         this.evaluatedProposalList = evaluatedProposalList;
         this.decryptedBallotsWithMeritsList = decryptedBallotsWithMeritsList;
+
+        txMap = blocks.stream()
+                .flatMap(block -> block.getTxs().stream())
+                .collect(Collectors.toMap(Tx::getId, Function.identity(), (x, y) -> y, HashMap::new));
```

Comment:
I assume the mergeFunction is only passed as you want to have the mapFactory. Not sure what the mergeFunction really should do as conflicts are not expected and not clear how to handle it. Maybe throwing an exception would be more appropriate here? Or maybe just add a comment.

Reply:
I wanted to guarantee that the output is a HashMap and it looks like the only `toMap` overload with a mapFactory param also has a mergeFunction param, so I was forced to provide a dummy merge function.

Reply:
Yes, I expected that intention...

I just was wondering how we can be sure to not change behaviour. The previous code used the flatMap. 

```
 public Stream<Tx> getTxStream() {
        return getBlocks().stream()
                .flatMap(block -> block.getTxs().stream());
    }
```

Do you know how potential key conflics would have been handled there? i assume your mergeFunction to overwrite with a new value if it happens is likely the standad behaviour if not otherwise defined. So your mergeFunction is likely better than throwing an exception if flatMap behaves the same. Anyway a bit "esoteric" but the DAO might deserve a bit of extra paranoia ;-)

Reply:
With the `(x, y) -> y` merge function it's currently using, it will always select the last tx with a given txId from the tx stream when building the map, whereas in the original code `getTx` is calling `findAny` on the filtered `getTxStream` output, which will probably behave the same as `findFirst` in this case. So perhaps the merge function should be `(x, y) -> x` to be absolutely sure the behaviour doesn't change.

Also, for consistent merge behaviour, `putIfAbsent` would need to be substituted into the line:

> `daoState.getTxMap().put(tx.getId(), tx);`

in `DaoState.onNewTxForLastBlock`.